x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/AllocateAction.java

```diff
@@ -88,7 +88,11 @@ public AllocateAction(
         } else {
             this.require = require;
         }
-        if (this.include.isEmpty() && this.exclude.isEmpty() && this.require.isEmpty() && numberOfReplicas == null) {
+        if (this.include.isEmpty()
+            && this.exclude.isEmpty()
+            && this.require.isEmpty()
+            && numberOfReplicas == null
+            && totalShardsPerNode == null) {
             throw new IllegalArgumentException(
                 "At least one of "
                     + INCLUDE_FIELD.getPreferredName()
```

Comment:
Yeah, I agree. This would be a breaking change though as it propagates all the way to the REST response. I don't think we should make this change as part of this PR (if we do want to make the breaking change). What do you think?

Reply:
I think it's only narrowly breaking (as in, every change is a breaking change in some way), and is extremely confusing for a user since it's incorrect. It's a bugfix to fix this message to me, so I think we should fix it.

(Also, no client, internally or externally, should be relying on specific strings in the message of an exception, as we make changes there all the time and they aren't really subject to BWC concerns)

Reply:
Update the error message in https://github.com/elastic/elasticsearch/pull/81944/commits/4a2f6ca73c0ba46477a63d80e934509b22e61bf4

Is this ok?

Reply:
That looks good to me, thanks for fixing it